### DISCUSSION TITLE
Redistribution of certain config variables: system settings vs theming.

### DISCUSSION THREAD

**Discussion by druncopsyken** [hyprwm/Hyprland#7997] [General]
Hey all,

I've been thinking that Hyprland has a good potential to implement a theming standard (sharing a file type for different looks). The issue as I see it now is that a few variables are put into different categories, while proper ones exist. I'm mainly looking at general, decorations and blur. Maybe some of the Animations as well. If some of the variables get shifted to the different categories, I think there could be theme files made where you only put in the complete categories of decorations, blur and animation. And the general category is more aimed towards what is enabled specific to the system.

Under general, there's the border_size,  col.* and gaps_in/out variables. I think these would be more suited to go under the decorations section. Then under the blur category, there's the variable if it's enabled. This also goes for animations. I think these should be moved to the general category as this section is best suited where the user/system owner decides how the compositor is to behave. Anything else that has to do with (global) coloring, should be more suited under decorations.

This might break quite some configs initially, but I think that would ease up a lot on later iterations as it gets adapted. Maybe someone will even design a pure theming applications to generate different themes to be shared.

Edit: I forgot to mention 3 layout variables. while the layout types differ from person to person, they do share 3 variables. smart_resizing, special_scale_factor and no_gaps_when_only. These might also be more suited to be put under the general category as they are shared and define system behavior instead of just layout specific.

**Comment by fufexan** [comment]
> Then under the blur category, there's the variable if it's enabled. This also goes for animations. I think these should be moved to the general category as this section is best suited where the user/system owner decides how the compositor is to behave.

I don't see why. It's a separate section, and it makes sense not to clutter the `general` section. Besides, it's not required to set `enabled = true;` in a "theme file", and `source`ing a theme wouldn't override the user's preference.

I agree with the rest though, a config reorganization would be nice.
